### PR TITLE
feat(BlockchainWallet): add function to get note placeholder

### PR DESCRIPTION
**Pull request by woudini**
Added getter for note placeholder. Similar to this function, but avoids unnecessary and time-consuming address derivation and instead matches indexes with labels:

https://github.com/blockchain/My-Wallet-V3-Frontend/blob/4e91f82e9e2bf1a235dacf9dd47348983a7796f1/assets/js/directives/transaction-note.directive.js#L23-L39

### CODE REVIEW

**Comment by woudini** [comment]
It looks like we may need to cherry-pick @Sjors ' recent dependency fixes

**Comment by Sjors** [comment]
You can just use `git merge master` to get the latest master fixes in this branch. But you didn't change an existing function, so it's safe to assume you didn't break any tests.

**Comment by Sjors** [comment]
Adding @jtormey and @plondon since they wrote the address lookup code for the frontend.

File: src/blockchain-wallet.js [hunk 1]
```diff
@@ -729,6 +729,33 @@ Wallet.prototype.deleteNote = function (txHash) {
...
+      var addresses = tx.processedOutputs.filter(function (processedOutput) { return processedOutput.identity >= 0 });
+      var indexesAndLabels = addresses.map(function (processedOutput) {
+        var coinType = processedOutput.coinType.split('/')
+        var addressIndex = coinType[2];
+        return {index: addressIndex, label: processedOutput.label};
+      });
+
+      if (addresses.length) {
+        if (isNaN(account)) account = addresses[0].identity;
+        var hdAddresses = MyWallet.wallet.hdwallet.accounts[account].receivingAddressesLabels;
```
Comment (Sjors) [inline src/blockchain-wallet.js]: Can't you use `this.hdwallet`?

**Comment by Sjors** [comment]
I'm not entirely sure what this code is doing. My approach would be to use `xpub":{"path":"M\/0\/1"` from `multiaddr` to get the address index.

Can you add some tests for this? Also can you make sure it doesn't break for a legacy wallet, before upgrade?

File: src/blockchain-wallet.js (hunk 1 above)
Comment (woudini) [inline src/blockchain-wallet.js]: Yes, changing now.

**Comment by woudini** [comment]
Added some comments and tests. It is getting the address index from the processed outputs, which have the paths in the same format in `coinType`.

If everything looks good, can I get another backlog of v3.22.13 with this commit to use for iOS?

**Comment by Sjors** [comment]
I'll make a new release based on `v3.21.2` with this commit added. That doesn't have the `tryLoadExternal` code.

**Pull request merged**